### ISSUE TITLE
Feature Request: `stack` function for masked arrays

### ISSUE THREAD

**Issue by lzkelley** [numpy/numpy#10127]
I was (perhaps naively) surprised to find that using `np.stack` on `masked_array`s returns a normal array.  `numpy.ma` provides the `vstack` and `hstack` methods, but not the generalized [`stack` method](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.stack.html) (and also the docstring which says to use `stack` instead of `vstack`/`hstack`).

I don't know how anything works, but the source for `numpy.ma.vstack` seems to suggest that it just wraps two calls to `numpy.vstack`, once for the data-arrays and once for the mask-arrays (see below), presumably the exact same thing could be done for `numpy.stack`.  Is it only a matter of adding `stack` to the list of functions which should be handled this way?

```
class _fromnxfunction_seq(_fromnxfunction):
    """
    A version of `_fromnxfunction` that is called with a single sequence
    of arrays followed by auxiliary args that are passed verbatim for
    both the data and mask calls.
    """
    def __call__(self, x, *args, **params):
        func = getattr(np, self.__name__)
        _d = func(tuple([np.asarray(a) for a in x]), *args, **params)
        _m = func(tuple([getmaskarray(a) for a in x]), *args, **params)
        return masked_array(_d, mask=_m)
```

Aside: it seems like it would be worth raising a warning when the returned array is *unmasked*, is that not a standard practice?

**Comment by eric-wieser** [comment]
>  Is it only a matter of adding stack to the list of functions which should be handled this way?

Make a PR and see, I suppose! Seems likely to me.

There's been talk of adding an `__array_concatenate__` which would allow `np.ma` to automatically work for any function that calls `concatenate` under the hood. That would work here too, but is still on the distant horizon

**Comment by lzkelley** [comment]
@eric-wieser can you point me towards this list in the src?

**Comment by eric-wieser** [comment]
@lzkelley: Searching for `_fromnxfunction_seq` in the same file as where you found it should show you this list.

**Comment by lzkelley** [comment]
@eric-wieser it looks like it was indeed as easy as suspected!  PR (https://github.com/numpy/numpy/pull/10223) checks out.

**Comment by eric-wieser** [comment]
Fixed in #10223
